jOOL-java-8/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -979,4 +979,694 @@ else if (percentile == 1.0)
+                        intercept = a.sumY / a.n - (a.sumXY - a.sumX * a.sumY / a.n) / (a.sumXX - a.sumX * a.sumX / a.n) * a.sumX / a.n;
+                    }
+
+                    Double avgY = a.sumY / a.n;
+                    Double sstot = 0.0;
+                    Double ssreg = 0.0;
+                    for (int i = 0; i < a.listY.size(); i++) {
+                        sstot += Math.pow((a.listY.get(i) - avgY), 2);
+                        ssreg += Math.pow(slope * a.listX.get(i) + intercept - a.listY.get(i), 2);
+                    }
+                    if (sstot == 0.0) {
+                        return Optional.of(1.0);
+                    } else {
+                        return Optional.of(1 - ssreg / sstot);
+                    }
+
+                }
+        );
+    }
+
+
+    //CS304 Issue link: https://github.com/jOOQ/jOOL/issues/151
+
+    /**
+     * Calculate the linear regression function Count of the given object collectors,
+     * based on the mapping function from object to double number.
+     *
+     * <p><pre> Usage of aggregation function regrCountBy():
+     * The mapping functions functionX and functionY are functions mapping the objects to double values X and Y, for instance:
+     * {@code
+     * FunctionX<Tuple2<Integer,Integer>, Double> mapping = e -> Double.valueOf(e.v1);
+     * FunctionY<Tuple2<Integer,Integer>, Double> mapping = e -> Double.valueOf(e.v2);
+     * }
+     * The specific usage of varianceBy is like:
+     * {@code
+     * Seq.of(new Tuple2<Integer,Integer>(1,1),new Tuple2<Integer,Integer>(2,2)).collect(Agg.regrCountBy(functionX,functionY));
+     * }
+     * Besides, self defined class is also allowed with mapping function:
+     * {@code
+     * Seq.of(new Node(-1,1),new Node(1,1)).collect(Agg.regrCountBy(functionX,functionY));
+     * }
+     * </pre>
+     *
+     * @param functionX mapping function from object to double value representing x
+     * @param functionY mapping function from object to double value representing y
+     * @return the Count value
+     * @version 1.0
+     * @author Jichen Lu
+     * @date 2021-05-17
+     */
+    public static <T> Collector<T, ?, Optional<Double>> regrCountBy(Function<? super T, ? extends Number> functionX, Function<? super T, ? extends Number> functionY) {
+        class Accumulator {
+            Integer n = 0;
+        }
+
+        return Collector.of(
+                () -> new Accumulator(),
+                (a, t) -> {
+                    Double x = (Double) functionX.apply(t);
+                    Double y = (Double) functionY.apply(t);
+                    if (x != null && y != null) {
+                        a.n++;
+                    }
+                },
+                (a1, a2) -> {
+                    a1.n += a2.n;
+                    return a1;
+                },
+                a -> {
+                    return Optional.of(Double.valueOf(a.n));
+                }
+        );
+    }
```

Comment:
For example (as mentioned before), this implementation could have certainly been done via a `Agg.filter(v -> v != null, Agg.count())` collector?